Source/AwsCommonRuntimeKit/crt/ByteBuffer.swift

```diff
@@ -2,285 +2,69 @@
 //  SPDX-License-Identifier: Apache-2.0.
 
 import struct Foundation.Data
-import class Foundation.InputStream
-import class Foundation.FileManager
-import class Foundation.FileHandle
-import class Foundation.OutputStream
-import struct Foundation.URL
 import AwsCIo
-import AwsCCommon
 import AwsCCal
-#if os(Linux)
-     import Glibc
- #else
-     import Darwin
- #endif
 
-// swiftlint:disable identifier_name superfluous_disable_command
-// Todo: this is being used. Refactor
-public class ByteBuffer: Codable {
+public class ByteBuffer {
+
+    private var data: Data
+    private var currentIndex: Data.Index
 
     public init(size: Int) {
-        array.reserveCapacity(size)
-        self.capacity = size
+        data = Data(capacity: size)
+        currentIndex = data.startIndex
     }
 
     public init(bytes: [UInt8]) {
-        self.array = bytes
-        self.capacity = bytes.count
-    }
-
-    //Todo: do not expose pointer
-    public init(ptr: UnsafeMutablePointer<UInt8>, len: Int, capacity: Int) {
-        let buffer = UnsafeBufferPointer(start: ptr, count: len)
-        self.array = Array(buffer)
-        self.capacity = capacity
-    }
-
-    public func allocate(_ size: Int) {
-        array = [UInt8]()
-        array.reserveCapacity(size)
-        self.capacity += size
-        currentIndex = 0
-    }
-
-    public func nativeByteOrder() -> Endianness {
-        return hostEndianness
+        self.data = Data(bytes)
+        currentIndex = data.startIndex
     }
 
-    public func currentByteOrder() -> Endianness {
-        return currentEndianness
+    public init(data: Data) {
+        self.data = data
+        currentIndex = data.startIndex
     }
 
-    public func order(_ endianness: Endianness) -> ByteBuffer {
-        currentEndianness = endianness
-        return self
+    public init(bufferPointer: UnsafeMutablePointer<UInt8>, length: Int, capacity: Int) {
+        self.data = Data(bytes: bufferPointer, count: capacity)
+        currentIndex = data.startIndex
     }
 
-    public func putByte(_ value: UInt8) {
-        array.append(value)
+    public func put(_ value: UInt8) {
+        data.append(value)
     }
 
-    public func put(_ value: UInt8) -> ByteBuffer {
-        array.append(value)
-        return self
+    public func put(_ value: [UInt8]) {
+        data.append(contentsOf: value)
     }
```

Comment:
Returing self is not needed https://github.com/awslabs/aws-crt-swift/pull/9#discussion_r526556883